Ventivader/Ventivader/Views/ProfileFieldView.swift

```diff
@@ -0,0 +1,141 @@
+        DispatchQueue.main.async { [weak self] in
+            self?.commonInit()
+            
+            guard let strongSelf = self else { return }
+            strongSelf.unitsLabel.text = units
+            strongSelf.unitsLabel.isHidden = units == nil
+            strongSelf.fieldTitleLabel.text = title
+            strongSelf.valueTextField.text = initialValue
+            
+            if type == .pastDate {
+                strongSelf.valueTextField.setInputViewDatePicker(target: strongSelf,
+                                                                 selector: #selector(strongSelf.dateSelected(sender:datePicker:)))
+            } else if type == .numeric {
+                strongSelf.valueTextField.keyboardType = .numberPad
+            } else if let options = options, options.count > 0 {
+                self?.pickerViewOptions = options
+                let pickerView = UIPickerView()
+                pickerView.delegate = self
+                strongSelf.valueTextField.inputView = pickerView
+            }
+        }
+    }
+    
+    private func commonInit() {
+        backgroundColor = ColorPallete.secondaryBackgroundColor
+        
+        fieldTitleLabel.textColor = ColorPallete.highlightColor
+        fieldTitleLabel.updateFont(name: VentivaderFonts.titleFont)
+        
+        unitsLabel.textColor = ColorPallete.highlightColor
+        unitsLabel.updateFont(name: VentivaderFonts.bodyFont)
+        setupTextField()
+    }
+    
+    func setupTextField() {
+        valueTextField.delegate = self
+    }
+    
+    @objc func dateSelected(sender: Any, datePicker: UIDatePicker) {
+        print(datePicker)
+        if let datePicker = valueTextField.inputView as? UIDatePicker {
+            let dateformatter = DateFormatter()
+            dateformatter.dateStyle = .medium
+            self.valueTextField.text = dateformatter.string(from: datePicker.date)
+        }
+        self.valueTextField.resignFirstResponder()
+    }
+}
+
+extension ProfileFieldView: UIPickerViewDelegate {
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return pickerViewOptions?[row] ?? ""
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        DispatchQueue.main.async { [weak self] in
+            self?.valueTextField.text = self?.pickerViewOptions?[row]
+        }
+    }
+}
+
+extension ProfileFieldView: UIPickerViewDataSource {
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return pickerViewOptions?.count ?? 0
+    }
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+}
+
+
+extension ProfileFieldView: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+        return true
+    }
+}
+
+extension UITextField {
```

Comment:
wonder if this should be in a file of its own instead of ProfileFieldView.swift as it is a UITextField extension. WDYT?

Reply:
Of course! Done.
I did the same for an UIStackView extension that was in VentilatorControlPanelViewController